### PR TITLE
Correct Delta protocol terminology "enabled" vs "supported"

### PR DESCRIPTION
**Pull request by xupefei**
## Description

This PR fixes a terminology issue in Delta protocol, so the term `supported` is now used to describe a table feature name being listed in table protocol's `readerFeatures` and/or `writerFeatures`. The choice of this word is to emphasize that, in such a scenario, the Delta table *may* use the listed table features but is not forced to do so.

For example, when `appendOnly` is listed in a table's protocol, the table may or may not be append-only, depending on the existence and value of table property `delta.appendOnly`. However, writers must recognize the table feature `appendOnly` and know that the table property should be checked before writing this table.

This PR did not touch the Row ID/Row Tracking sections, as it's handled by another PR: https://github.com/delta-io/delta/pull/1747.

### CODE REVIEW

**Comment by xupefei** [comment]
@vkorukanti Can you review this PR?

File: PROTOCOL.md [hunk 1]
```diff
@@ -700,18 +705,21 @@ For example, given a table on Reader Version 1 and Writer Version 4, along with
...
 
-A feature being enabled does not imply that it is active. For example, a table may have the [Append-only Tables](#append-only-tables) feature (feature name `appendOnly`) enabled in `writerFeatures`, but does not satisfy a table property `delta.appendOnly` equals to `true`. In such a case the table is not append-only, and writers are allowed to change, remove, and rearrange data. However, writers must implement the feature to know that the table property `delta.appendOnly` should be checked.
+A feature being supported does not imply that it is active. For example, a table may have the [Append-only Tables](#append-only-tables) feature (feature name `appendOnly`) listed in `writerFeatures`, but does not satisfy a table property `delta.appendOnly` equals to `true`. In such a case the table is not append-only, and writers are allowed to change, remove, and rearrange data. However, writers must know that the table property `delta.appendOnly` should be checked before writing the table.
+
+## Enabled Features
```
Comment (larsk-db) [inline PROTOCOL.md:713]: Should this be called `active` then, consistent with the usage in the previous paragraph?

File: PROTOCOL.md [hunk 2]
```diff
@@ -700,18 +705,21 @@ For example, given a table on Reader Version 1 and Writer Version 4, along with
...
 
 ## Disabled Features
-A feature is `disabled` if it is in neither `readerFeatures` nor `writerFeatures`. Writers are allowed to `enable` a feature for the table by adding its name to the `readerFeatures` or `writerFeatures`. Reader features should be added to both `readerFeatures` and `writerFeatures` simultaneously, while writer features should be added only to `writerFeatures`. It is not allowed to add features only to `readerFeatures` but not to `writerFeatures`.
+A feature is `disabled` if it is in neither `readerFeatures` nor `writerFeatures`. Writers are allowed to add support of a feature to the table by adding its name to `readerFeatures` or `writerFeatures`. Reader features should be listed in both `readerFeatures` and `writerFeatures` simultaneously, while writer features should be listed only in `writerFeatures`. It is not allowed to list a feature only in `readerFeatures` but not in `writerFeatures`.
```
Comment (larsk-db) [inline PROTOCOL.md:717]:
```suggestion
A feature is `disabled` if it is neither in `readerFeatures` nor in `writerFeatures`. Writers are allowed to add support of a feature to the table by adding its name to `readerFeatures` or `writerFeatures`. Reader features should be listed in both `readerFeatures` and `writerFeatures` simultaneously, while writer features should be listed only in `writerFeatures`. It is not allowed to list a feature only in `readerFeatures` but not in `writerFeatures`.
```

File: PROTOCOL.md [hunk 3]
```diff
@@ -700,18 +705,21 @@ For example, given a table on Reader Version 1 and Writer Version 4, along with
...
+A feature being supported does not imply that it is active. For example, a table may have the [Append-only Tables](#append-only-tables) feature (feature name `appendOnly`) listed in `writerFeatures`, but does not satisfy a table property `delta.appendOnly` equals to `true`. In such a case the table is not append-only, and writers are allowed to change, remove, and rearrange data. However, writers must know that the table property `delta.appendOnly` should be checked before writing the table.
+
+## Enabled Features
+A feature is enabled by a table when it is supported, plus it's metadata requirements are satisfied. Each feature defines its own metadata requirements, as stated in corresponding sections of this document. For example, Append-only feature is enabled when there exists `appendOnly` feature name in a `protocol`'s `writerFeatures`, plus a table property `delta.appendOnly` set to `true`.
 
 ## Disabled Features
```
[the remark this answers is not in the document]
Comment (xupefei) [reply PROTOCOL.md:716]: Done!

File: PROTOCOL.md [hunk 4]
```diff
@@ -763,10 +769,10 @@ In `id ` mode, readers must resolve columns by using the `field_id` in the parqu
...
 
 # Deletion Vectors
-Enablement:
- - To enable Deletion Vectors on a table, the table must have Reader Version 3 and Writer Version 7. A feature name `deletionVectors` must exist in the table's `readerFeatures` and `writerFeatures`.
+To support this feature:
+ - To support Deletion Vectors, a table must have Reader Version 3 and Writer Version 7. A feature name `deletionVectors` must exist in the table's `readerFeatures` and `writerFeatures`.
 
-When enabled:
+When supported:
  - A table's `add` and `remove` actions can optionally include a Deletion Vector (DV) that provides information about logically deleted rows, that are however still physically present in the underlying data file and must thus be skipped during processing. Readers must read the table considering the existence of DVs.
```
Comment (bart-samwel) [inline PROTOCOL.md:805]:
```suggestion
 - A table's `add` and `remove` actions can optionally include a Deletion Vector (DV) that provides information about logically deleted rows, that are however still physically present in the underlying data file and must thus be skipped during processing. Readers must read the table considering the existence of DVs, even when the `delta.enableDeletionVectors` table property is not set.
```

File: PROTOCOL.md [hunk 5]
```diff
@@ -763,10 +769,10 @@ In `id ` mode, readers must resolve columns by using the `field_id` in the parqu
 In `name` mode, readers must resolve columns in the data files by their physical names as given by the column metadata property `delta.columnMapping.physicalName` in the Delta schema. Partition values and column level statistics will also be resolved by their physical names. For columns that are not found in the files, `null`s need to be returned. Column ids are not used in this mode for resolution purposes.
 
 # Deletion Vectors
-Enablement:
- - To enable Deletion Vectors on a table, the table must have Reader Version 3 and Writer Version 7. A feature name `deletionVectors` must exist in the table's `readerFeatures` and `writerFeatures`.
+To support this feature:
+ - To support Deletion Vectors, a table must have Reader Version 3 and Writer Version 7. A feature name `deletionVectors` must exist in the table's `readerFeatures` and `writerFeatures`.
 
-When enabled:
+When supported:
```
Comment (bart-samwel) [inline PROTOCOL.md:804]: I think there's something missing in the protocol about _activating_ or _enabling_ DVs. AFAICT there is no mention of `delta.enableDeletionVectors`. Can you fix that as well?

**Comment by felipepessoto** [comment]
May need to update the changes from #1780

File: PROTOCOL.md (hunk 5 above)
Comment (xupefei) [reply PROTOCOL.md:804]: Done!

**Pull request closed without merging**